### ISSUE TITLE
websocket: Unable to hijack the connection

### ISSUE THREAD

**Issue by yuriy-zhilovets2** [containous/traefik#1315]
`v1.2.0-rc2`
`v1.2.0`

etcd backend on Debian box

Configuration (read from etcd):
```
etcdctl get /traefik/backends/r-announcer/servers/server41/url
ws://172.25.255.41:2010

etcdctl get /traefik/frontends/r-announcer/passHostHeader
false

etcdctl get /traefik/frontends/r-announcer/backend
r-announcer

etcdctl get /traefik/frontends/r-announcer/entrypoints
https

etcdctl get /traefik/frontends/r-announcer/routes/host/rule
Host:announcer.mol.net.ua

etcdctl get /traefik/entrypoints/https/address
:443

etcdctl get /traefik/entrypoints/https/compress
false

tls settngs skipped because they work with https wihout any problems

```

```
time="2017-03-22T21:05:47+03:00" level=debug msg="Creating frontend r-announcer" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Wiring frontend r-announcer to entryPoint http" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Creating route host Host:announcer.mol.net.ua" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Wiring frontend r-announcer to entryPoint https" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Creating route host Host:announcer.mol.net.ua" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Creating backend r-announcer" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Creating load-balancer wrr" 
time="2017-03-22T21:05:47+03:00" level=debug msg="Creating server server41 at ws://172.25.255.41:2010 with weight 0"
```

The settings for proxiyng web sockets are fairly standard. I have studied thoroughly https://github.com/containous/traefik/issues/452, so there are no:

ddr balancer (on this backend)
circuitbreaker (on this backend)
retry (globally)
compress (on entrypoint https)

But all log is filled with:

> Unable to hijack the connection: *negroni.responseWriter the ResponseWriter doesn't support the Hijacker interface

and all connections to frontend URL immediately return 500 without any attempts to call backend.

**Comment by thomasf** [comment]
I see this problem as well after upgrading to 1.2 from 1.1.2

**Comment by emilevauge** [comment]
Could give a try without retry and then without circuit breaker ?

**Comment by thomasf** [comment]
I have none of those enabled if I understand things correctly..

I just have a single back end (the latest offifical mattermost docker images) configured via docker.

Retry or Compress is not mentioned in my traefik toml config file.

The status page says: 'Load Balancer: wrr'

my docker-compose-file:

```
 labels:
      traefik.backend: mattermost
      traefik.port: "80"
      traefik.frontend.rule: Host:somehost.com
      traefik.docker.network: traefik
      traefik.enable: "true"
```

**Comment by yuriy-zhilovets2** [comment]
That's my results without any circuitbreaker and retry is not mentioned in my backend.

**Comment by Vincoll** [comment]
I also encounter this error systematically, causing a 500 error on websockets.
If necessary I can provide details logs and informations for the bug fix.


### What version of Traefik are you using?
```
Version:      v1.2.1
Codename:     morbier
Go version:   go1.7.5
Built:        2017-03-27_03:19:40PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml...)?
```
{
 "GraceTimeOut": 10,
 "Debug": false,
 "CheckNewVersion": true,
 "AccessLogsFile": "",
 "TraefikLogsFile": "",
 "LogLevel": "ERROR",
 "EntryPoints": {
  "http": {
   "Network": "",
   "Address": ":80",
   "TLS": null,
   "Redirect": {
    "EntryPoint": "https",
    "Regex": "",
    "Replacement": ""
   },
   "Auth": null,
   "Compress": false
  },
  "https": {
   "Network": "",
   "Address": ":443",
   "TLS": {
    "MinVersion": "",
    "CipherSuites": null,
    "Certificates": [
     {
      "CertFile": "-----BEGIN CERTIFICATE-----[...]-----END CERTIFICATE-----",
      "KeyFile": "-----BEGIN PRIVATE KEY-----[...]-----END PRIVATE KEY-----"
     }
    ],
    "ClientCAFiles": null
   },
   "Redirect": null,
   "Auth": null,
   "Compress": false
  }
 },
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [
  "http",
  "https"
 ],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "InsecureSkipVerify": false,
 "Retry": null,
 "Docker": null,
 "File": null,
 "Web": {
  "Address": ":8080",
  "CertFile": "",
  "KeyFile": "",
  "ReadOnly": false,
  "Statistics": {
   "RecentErrors": 50
  },
  "Metrics": null,
  "Auth": {
   "Basic": {
    "Users": [
     "[...]:[...]"
    ]
   },
   "Digest": null,
   "HeaderField": ""
  }
 },
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": {
  "Watch": true,
  "Filename": "",
  "Constraints": null,
  "Endpoint": "",
  "AccessKey": "",
  "SecretKey": "",
  "ExposedByDefault": false,
  "Domain": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
 },
 "ConfigFile": "/etc/traefik/traefik.toml"
}
```

**Comment by blackbarn** [comment]
Similar problem.

`time="2017-04-09T00:37:07Z" level=info msg="Traefik version v1.2.1 built on 2017-03-27_03:19:40PM"`

Using docker. 

I've got a container specifying a label `traefik.frontend.entryPoints=https,wss`

https works fine, just the trouble with websockets.

**Comment by Ranger-X** [comment]
Try to turn off detailed statistics (subsection [web.statistics] in [web]). Unexpected, but it helped me.

**Comment by yuriy-zhilovets2** [comment]
Unbelievable but true. Turning off the statistics fixes the error. Thank you, Ranger-X.

**Comment by Vincoll** [comment]
I confirm too. WS are OK now.
Thank you !

**Comment by emilevauge** [comment]
Hum, thanks for this investigation @Ranger-X.
We need to add to `StatsRecorder`:
```
var (
	_ http.ResponseWriter = &StatsRecorder{}
	_ http.Hijacker       = &StatsRecorder{}
	_ http.Flusher        = &StatsRecorder{}
	_ http.CloseNotifier  = &StatsRecorder{}
)
```

**Comment by phoenix741** [comment]
I have the same problem, if i comment this two line like this :
#  [web.statistics]
#  RecentErrors = 10

The problem is resolved too.

**Comment by ldez** [comment]
Closed by #1598
